types/coin.go

```diff
@@ -200,8 +200,15 @@ func (coins Coins) IsValid() bool {
 		}
 
 		lowDenom := coins[0].Denom
+		seenDenoms := make(map[string]bool)
+		seenDenoms[strings.ToUpper(lowDenom)] = true
+
 		for _, coin := range coins[1:] {
-			if strings.ToLower(coin.Denom) != coin.Denom {
```

Comment:
I'm pretty sure this was a bug

Reply:
fix godoc here:  https://github.com/cosmos/cosmos-sdk/pull/7027/files#diff-17d06f54275a8b920aeab6538617c7bdR186